File: 406-根据身高重建队列.py
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        people.sort(key=lambda x: x[1])
        people.sort(key=lambda x: x[0], reverse=True)
        result = []

        for i in range(len(people)):
            result.insert(people[i][1], people[i])
        
        return result

File: 406-根据身高重建队列.cpp
class Solution {
public:
    static bool compare(const vector<int>& a, const vector<int>& b){
        if(a[0] > b[0] || (a[0] == b[0] && a[1] < b[1]))
            return true;
        return false;
    }

    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
        stable_sort(people.begin(), people.end(), compare);

        vector<vector<int>> result;
        
        for(int i = 0; i < people.size(); i++){
            result.insert(result.begin() + people[i][1], people[i]);
        }

        return result;
    }
};
